addons/project/models/project_task.py

```diff
@@ -937,13 +938,39 @@ def create(self, vals_list):
             # if the portal user could really create the tasks based on the ir rule.
             tasks.with_user(self.env.user).check_access_rule('create')
         current_partner = self.env.user.partner_id
+
+        all_partner_emails = []
+        for task in tasks:
+            all_partner_emails += tools.email_split(task.email_cc)
+        partners = self.env['res.partner'].search([('email', 'in', all_partner_emails)])
+        partner_per_email = defaultdict()
+        for partner in partners:
+            if not all(u.share for u in partner.user_ids):
+                partner_per_email[partner.email] = partner
         for task in tasks:
             if task.project_id.privacy_visibility == 'portal':
                 task._portal_ensure_token()
             for follower in task.parent_id.message_follower_ids:
                 task.message_subscribe(follower.partner_id.ids, follower.subtype_ids.ids)
             if current_partner not in task.message_partner_ids:
                 task.message_subscribe(current_partner.ids)
+            if task.email_cc:
+                partners_with_internal_user_ids = []
+                for email in tools.email_split(task.email_cc):
+                    if partner_per_email.get(email, False):
+                        partners_with_internal_user_ids += [partner_per_email[email].id]
+                if not partners_with_internal_user_ids:
+                    continue
+                task.message_subscribe(partners_with_internal_user_ids)
+                message_values = {
+                    'subject':  Markup(f'Invitation to follow Task: {task.name}'),
+                    'body': Markup(f'<p>Hello,</p><p>You have been invited to follow Task document: {task.name}</p>'),
+                    'record_name':  Markup(task.name),
```

Comment:
Is it really needed to use Markup here? 🧐

Reply:
I'm not really sure to understand in which case markup is needed or not. My understanding is that is is used to ensure input from the user is safe to be used for the construction of html/xml pages. So here's my train of thougths : in this case, we are using the name of the task (which is an input from the user) to construct the mail that's going to be sent to internal user. While i suppose that the name has passed some sort of check up to ensure it was safe for the db (since the tasks are already created once this part of the create method is reached), I don't think that that check up also ensures it's safe html wise. And since the mail can be accessed with the technical/emails menu, I left the markup.

Reply:
For me, the Markup function is only useful when we have html in the string and indeed you right and since we cannot really be sure task name does not contain html the Markup function could be useful, it is better to have it even if it could be useless in 90% rather than having a security issue. 🙂